### PR TITLE
flashcard service:  dapr topic and deletion part

### LINKED ISSUE #19: Flashcard service: Delete flashcards when content is deleted
## Description
The flashcard service needs to listen to content change events for deletion operations.
When a deletion event occurs, all flashcards linked to the (in the event specified) deleted contents need to also be deleted. 

## Acceptance Criteria
- [ ] flashcard service listens to dapr topic for content changes
- [ ] service deletes all flashcards linked to deleted content

### PR DESCRIPTION
**Pull request by Keyshal-211**
service deletes all flashcards linked to deleted content

## Description of changes
Please provide a short description of the change or if the change is sufficiently described in the GitHub issue, provide a link to the issue.

  
## How has this been tested?

Please describe the test strategy you followed. If not tested, please explain why.

- [ ] automated unit test
- [ ] automated integration test
- [ ] manual, exploratory test

In case of manual test, please document the test well including a set of user instructions and prerequisites. Each including an action, it's result, and where appropriate a screenshot.
    
## Checklist before requesting a review

- [ ] My code is easy to understand
- [ ] My code follows the [coding guidelines](https://github.com/IT-REX-Platform/wiki/blob/main/dev-manuals/backend/coding-guidelines.md) of this project
- [ ] My code fulfills all acceptance criteria
- [ ] The test coverage (line and branch) is reasonably high, especially on Service classes and other classes with complex logic
- [ ] I have made corresponding changes to the documentation or
  the [wiki/adr](https://github.com/IT-REX-Platform/wiki/tree/main/adr)
- [ ] I made no breaking changes in the database schema or if so, I will perform a database migration

## Checklist for reviewer

- The code is easy to understand
- The code follows
  the [coding guidelines](https://github.com/IT-REX-Platform/wiki/blob/main/dev-manuals/backend/coding-guidelines.md) of
  this project
- The code is tested or if not, the reason is documented or discussed
- The added and existing tests reasonably cover the code change
- The code has no breaking changes in the database schema or if so, the assignee is aware of it and
  will perform a database migration

### CODE REVIEW

File: src/main/java/de/unistuttgart/iste/gits/flashcard_service/service/FlashcardService.java [hunk 1]
```diff
@@ -115,5 +117,41 @@ public List<FlashcardSet> getFlashcardSetsByAssessmentId(List<UUID> ids) {
...
+            log.error(e.getMessage());
+            return;
+        }
+        // only consider DELETE Operations
+        if (!dto.getOperation().equals(CrudOperation.DELETE) || dto.getContentIds().isEmpty()) {
+            return;
+        }
+
+        // find all quizzes
+        List<FlashcardEntity> flashcardEntities = flashcardRepository.findAllById(dto.getContentIds());
```
Comment (BlueWaves2) [inline src/main/java/de/unistuttgart/iste/gits/flashcard_service/service/FlashcardService.java]: This won't work. The content ID matches the FlashcardSET ID, not the flashcard ID

**Review by BlueWaves2 — CHANGES_REQUESTED** [review]
- needs to retrieve the flashcards via flashcard SET ID 
- tests are missing (Test coverage is 4.3%)

File: src/main/java/de/unistuttgart/iste/gits/flashcard_service/service/FlashcardService.java (hunk 1 above)
Comment (Keyshal-211) [reply src/main/java/de/unistuttgart/iste/gits/flashcard_service/service/FlashcardService.java]: done with testes

**Review by BlueWaves2 — APPROVED** [review]
looks fine now!

**Pull request merged**